src/context/DataContext.js

```diff
@@ -0,0 +1,22 @@
+import { createContext, useContext, useReducer } from "react";
+import { reducerFunc, initialState } from "../reducer/reducer";
+
+const DataContext = createContext();
+
+export const useData = () => useContext(DataContext);
+
+export const DataProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducerFunc, initialState);
```

Comment:
Its better to rename the function as `dataReducerFunction`, so as to avoid complexity when using multiple reducers.